src/components/Card.js

```diff
@@ -3,72 +3,128 @@
 // const cardElement = cardTemplate.cloneNode(true);
 
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(cardData,
+    cardSelector,
+    handleImageClick,
+    handleDeleteSubmitPopup,
+    handleCardLike,
+    checkLikeStatus,
+  ) {
+    this._cardData = cardData;
+    
+    this._name = cardData.name;
+    this._link = cardData.link;
+    this._cardId = cardData._id;
+    this._isLiked = cardData.isLiked;
     this._cardSelector = cardSelector;
-    //this._cardElement = this._getCardElement();
     this._handleImageClick = handleImageClick;
+    this._handleDeleteSubmitPopup = handleDeleteSubmitPopup;
+    this.handleCardLike = handleCardLike;
+    this.checkLikeStatus = checkLikeStatus;
+    
+    
   }
 
-  _setEventListeners() {
-    // this._likeButton.addEventListener("click", () => {
-    //   this._handleLikeIcon();
-    // });
+  getId() {
+    return this._cardId;
+  }
+  
+  toggleLikeIcon(cardData) {
+    
+    if (this._likeButton.classList.contains("cards__like-button_active")) {
+      this._likeButton.classList.remove("cards__like-button_active");
+      console.log(cardData, "is TOGGLED TO unliked");
+    } else {
+      this._likeButton.classList.add("cards__like-button_active");
+      console.log(cardData, "is TOGGLED TO liked");  
+    }}
 
-    // this._deleteButton.addEventListener("click", () => {
-    //   this._handleDeleteCard();
-    // });
+    
+    
+    submitButtonLoadingState(isLoading) {
```

Comment:
Please, delete `submitButtonLoadingState` from `Card` because you have it in `PopupWithForm`


You don't use it here and you can't use it